### DISCUSSION TITLE
Q&A for MP2

### DISCUSSION THREAD

**Discussion by stan6** [SOEN345-WINTER2024/discussion#22] [Q&A]
Q: `For the MP2 assignment for SOEN 345, are we allowed to modify the response from ChatGPT for the UI testing question to adapt it to our app better? Thank you.`

A: Thank you for the question. Yes, modifying the prompt to ChatGPT is fine to better adapt to the app. If you find better prompt design, you can also submit under https://github.com/orgs/SOEN345-WINTER2024/discussions/2

**Comment by jesseythach** [comment]
Can we modify the response after giving the prompt to ChatGPT?

**Reply by stan6** [reply]
Oh. you mean further interact with ChatGPT? Yes, you can.

**Comment by RyanGuzelian** [comment]
Is app A supposed to be the same app as from the project?

**Reply by stan6** [reply]
Yes. App A is your selected app for the project.

**Reply by RyanGuzelian** [reply]
The app i selected was not from the [repository provided in MP2](https://github.com/pcqpcq/open-source-android-apps/blob/master/README.md), it was from the [F-Droid main repository](https://apt.izzysoft.de/fdroid/index.php/?repo=main), which was an option when selecting the app for the project. F-Droid does not have categories. What should I do when selecting app B, which should be from the same category as app A?

**Reply by stan6** [reply]
You can choose category based on the functionality of your selected app.

**Comment by jesseythach** [comment]
For question 2, where we have to select app B from the link provided in the instructions, can we find it elsewhere as long as it's open-source instead of from that link?

**Reply by stan6** [reply]
You are free to select app B as long as (1) it is of different categories, and (2) use GitHub for issue tracking. But since you don't need to download nor run app B, you can just choose any app from the provided link to save time in searching for an app.

**Comment by jesseythach** [comment]
In the instructions, it says "All answers (except for the code modifications) should be written in README.md", but I didn't have to do any code modifications to any of the questions. What does it mean? Which question is that for? thank you.

**Reply by stan6** [reply]
Hi, thanks. It is a general comment. The code modifications is only if you record UI tests or fixing code for the bugs you found.

**Comment by AA789-ai** [comment]
For question 2, it asks us if we can reproduce the issue of app B in our app A. Does being unable to reproduce the issue using the same / similar behavior in our app A count as an explanation? If not, how should I explain the reason? Thank you so much.

**Reply by stan6** [reply]
Hi, we only expect simple reason like you cannot reproduce because the button doesn't exist, or the app behaves normally without displaying wrong output/crash.

**Reply by AA789-ai** [reply]
Thank you so much!

**Comment by AA789-ai** [comment]
For question 1.c, It asks us to reproduce UI tests in our selected apps based on chatGPT prompts. Do these tests have to be automated (ie, using espresso) or it can be a mix of automated and manual tests? Thank you so much.

**Reply by stan6** [reply]
You can just manually reproduce the same UI actions given by ChatGPT (e.g., clicking button A, then press button B...) in your app and see if the app gives you unexpected behavior.

**Reply by AA789-ai** [reply]
Thank you so much!

**Comment by stan6** [comment]
`Q: Could you please explain what you need from us in question 3 because I don’t understand what I suppose to do in it. As it says: "Have you found any bug using any of the above method?”, but we didn’t use any methods.` 

A: There are two methods (1) UI testing using ChatGPT, (2) collaborative bug finding using GitHub issues that you tried in previous questions. You need to compare the effectiveness of these methods and answer which one you think that is better?
